### PR TITLE
Location scoping 25 mile radius

### PR DESCRIPTION
**Pull request by shaniyaclement**
Implemented data constraints that will only query posts (ParseObject) created within a 25 mile radius of the current user's current location. To do this, I used the Connectivity Manger to establish whether there is wifi to run Google Play location services. If wifi, the application prompts the user to allow location access, if not already granted. Once granted, a location request is ran on a loop every 2 miles or 60 seconds using getFusedLocationProviderClient location updates. The application tracks the updates and assigns the user current location Parse as it changes. When queried, the application runs a Parse Query for the posts created in the surrounding area. To know what where the post is made, location tracking is added to the compose post activity, and the user location at the time of posting is added the a new post when it is saved.

Following this, I want to add user can view post of people they are following and querying based on user interest.

Problem2

### CODE REVIEW

File: app/src/main/java/com/example/journeyjournal/ParseConnectorFiles/Post.java [hunk 1]
```diff
@@ -46,14 +47,12 @@ public void setUser(ParseUser user) {
         put(KEY_USER, user);
     }
 
-    public ParseGeoPoint getLocation() {
-        return getParseGeoPoint(KEY_LOCATION);
-    }
-
     public void setLocation(ParseGeoPoint location) {
```
Comment (dengfengjiang) [inline app/src/main/java/com/example/journeyjournal/ParseConnectorFiles/Post.java:53]: remove this also

File: app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java [hunk 2]
```diff
@@ -147,11 +146,11 @@ protected void startLocationUpdates() {
         // Create the location request to start receiving updates
         LocationRequest mLocationRequest = LocationRequest.create();
         mLocationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
-        /* 10 secs */
-        long UPDATE_INTERVAL = 10 * 1000;
-        mLocationRequest.setInterval(UPDATE_INTERVAL);
-        /* 2 sec */
-        long FASTEST_INTERVAL = 2000;
+        /* 2 Mile */
+        long MIN_DISTANCE = 2 * 1609;
```
Comment (dengfengjiang) [inline app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java]: move defined constant at beginning of class.

**Review by dengfengjiang — APPROVED** [review]

**Review by dengfengjiang — APPROVED** [review]

File: app/src/main/java/com/example/journeyjournal/Activities/ComposePostActivity.java [hunk 3]
```diff
@@ -49,6 +60,19 @@ public class ComposePostActivity extends AppCompatActivity {
     ImageView ivImage;
     protected PostsAdapter adapter;
     protected List<Post> allPosts;
+    double latitude;
```
Comment (dengfengjiang) [inline app/src/main/java/com/example/journeyjournal/Activities/ComposePostActivity.java]: this is local parameter

**Review by dengfengjiang — APPROVED** [review]

**Review by dengfengjiang — APPROVED** [review]

**Review by dengfengjiang — APPROVED** [review]

File: app/src/main/java/com/example/journeyjournal/ParseConnectorFiles/Post.java (hunk 1 above)
Comment (shaniyaclement) [reply app/src/main/java/com/example/journeyjournal/ParseConnectorFiles/Post.java:53]: Removed
Screen Shot 2022-07-22 at 11 51 35 AM

File: app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java (hunk 2 above)
Comment (shaniyaclement) [reply app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java]: resolved

File: app/src/main/java/com/example/journeyjournal/Activities/ComposePostActivity.java [hunk 4]
```diff
@@ -89,15 +120,63 @@ public void onClick(View v) {
...
+        Toast.makeText(this, "Permission ok", Toast.LENGTH_SHORT).show();
 
     }
 
+    public void onLocationChanged(Location location) {
+        double latitude;
+        double longitude;
+        // New location has now been determined
+        latitude = location.getLatitude();
+        longitude = location.getLongitude();
```
Comment (davidlin54) [inline app/src/main/java/com/example/journeyjournal/Activities/ComposePostActivity.java:173]: can combine these

`double latitude = location.getLatitude();`
`double longitude = location.getLongitude();`

File: app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java [hunk 5]
```diff
@@ -133,18 +132,69 @@ public void onClick(View v) {
...
+    }
+
+    public void onLocationChanged(Location location) {
+        // New location has now been determined
+        latitude = location.getLatitude();
+        longitude = location.getLongitude();
+        String msg = "Updated Location: " +
+                latitude + "," +
+                longitude;
+        Toast.makeText(getActivity(), "on location changed", Toast.LENGTH_SHORT).show();
```
Comment (davidlin54) [inline app/src/main/java/com/example/journeyjournal/fragments/FeedFragment.java]: do you want to show 2 toasts here? maybe you can just remove this one

**Pull request merged**